src/custom.rs

```diff
@@ -103,3 +103,39 @@ pub fn getrandom_inner(dest: &mut [MaybeUninit<u8>]) -> Result<(), Error> {
         Some(code) => Err(Error::from(code)),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    const LEN7_CODE: u32 = Error::CUSTOM_START + 7;
+    // Returns a custom error if input is length 7, otherwise fills with 0x55.
+    fn insecure_rng(dest: &mut [u8]) -> Result<(), Error> {
+        if dest.len() == 7 {
+            return Err(NonZeroU32::new(LEN7_CODE).unwrap().into());
+        }
+        dest.fill(0x55);
+        Ok(())
+    }
+
+    // Test registering a custom implementation, even on supported platforms.
+    register_custom_getrandom!(insecure_rng);
+
+    // On a supported platform, make sure the custom implementation isn't used.
+    #[cfg(target_os = "linux")]
```

Comment:
I suggest use use:
```
cfg_if! {
    if #[cfg(we should be using custom implementation)] {
         #[test]
         fn custom_used() { ... }
    } else { 
        #[test]
         fn custom_not_used() { .... }
    }
}
```
In particular, we should run custom_not_used on all targets, not just Linux, right?

Reply:
I added a few additional OS targets (Windows and macOS) I don't think it's necessary to run them on _every_ supported target, checking a few common platforms seems fine.